internal/rbd/controllerserver.go

```diff
@@ -944,6 +944,20 @@ func (cs *ControllerServer) doSnapshotClone(ctx context.Context, parentVol *rbdV
 		return ready, cloneRbd, err
 	}
 
+	defer func() {
+		// If we hit any error below, cleanup the snapshot created on the clone
+		if err != nil {
```

Comment:
this if gets nested pretty deep. Is it needed to do the `err != nil` check before calling `errors.Is()`?